addons/account/models/account_payment.py

```diff
@@ -4,15 +4,46 @@
 from odoo.exceptions import UserError, ValidationError
 
 
-class AccountPaymentMethod(models.Model):
-    _name = "account.payment.method"
-    _description = "Payment Methods"
-    _order = 'sequence'
+class AccountPaymentMethodType(models.Model):
+    _name = "account.payment.method.type"
+    _description = "Method Types"
 
     name = fields.Char(required=True, translate=True)
     code = fields.Char(required=True)  # For internal identification
     payment_type = fields.Selection([('inbound', 'Inbound'), ('outbound', 'Outbound')], required=True)
-    sequence = fields.Integer(help='Used to order Methods in the form view', default=10)
+
+
+class AccountPaymentMethod(models.Model):
+    _name = "account.payment.method"
+    _description = "Payment Methods"
+
+    name = fields.Char(default='New method')
```

Comment:
Like discussed, define `name_get` to compute the `display_name` instead. (and probably `name_search` too)